internal/util/update/resource-merge.go

```diff
@@ -99,3 +107,35 @@ func (u ResourceMergeUpdater) updatedKptfile(updatedPath string, options UpdateO
 	err = kf.MergeOpenAPI(options.KptFile)
 	return kf, err
 }
+
+// setAllSetterDefinitions reads all the Setter Definitions from Kptfile in local
+// package path and sets all setter values in original and updated folders for
+// resource merge to be accurate
+func setAllSetterDefinitions(packagePath, originalPath, updatedPath string) error {
+	// get all the setter definitions
+	l := setters2.List{}
+	err := l.List(filepath.Join(packagePath, "Kptfile"), packagePath)
```

Comment:
Add a TODO that this could be optimized to perform all the setters in 1 pass instead of N passes

Reply:
Already done.